src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -152,14 +154,21 @@ void ConnectionHandlerImpl::OnDeviceAdded(
                 device_info.mac_address(),
                 device_info.connection_type());
 
-  auto result = device_list_.insert(std::make_pair(handle, device));
+  {
+    sync_primitives::AutoWriteLock write_lock(device_list_lock_);
+    auto result = device_list_.insert(std::make_pair(handle, device));
 
-  if (!result.second) {
-    LOG4CXX_ERROR(logger_,
-                  "Device with handle " << handle
-                                        << " is known already. "
-                                           "Information won't be updated.");
-    return;
+    if (!result.second) {
+      LOG4CXX_ERROR(logger_,
+                    "Device with handle " << handle
+                                          << " is known already. "
+                                             "Information won't be updated.");
+      return;
+    }
+  }
+  if (device_info.name() ==
+      transport_manager::webengine_constants::kWebEngineDeviceName) {
+    connection_handler_observer_->OnWebEngineDeviceCreated();
```

Comment:
Should the connection_handler_observer_lock_ should be used in this method?

Reply:
@JackLivio No. The reasons why: 

- OnWebEngineDeviceCreated() must be called only from one place in the code (and it is)
- As connection_handler_observer is application manager and inside function body it uses 3 function protected by separate locks:
1) policy_handler (protected by separate policy lock)
2) CreatePendingLocalApp uses inside critical section which is also protected: 
```cpp
  sync_primitives::AutoLock lock(apps_to_register_list_lock_ptr_);
  apps_to_register_.insert(application);
```
3) SendUpdateAppList uses current applications list to be sent to tht HMI.